### PR TITLE
feat: Lazily evaluated env template variables in path entries

### PR DESCRIPTION
**Pull request by josb**
Implements #1912.

Please note that there's a TODO entry in `mise::EnvMapVisitor.visit_map`. The issue is that `deserialize_with = "deserialize_arr"` can either be left off or added to`path: Vec<PathEntry>`:
* If it is left off, the feature works as demonstrated but 3 tests fail;
* if it is added, lazy path entries (`{ value = "{{ env.GEM_HOME }}/bin" }`) cause a parse error (`invalid type: map, expected a string`) but the 3 failing tests now pass.

I am hoping you can help me address this issue, as I don't understand Serde / Rust well enough yet.
This is also the reason why I didn't add any tests for this feature yet.

Thanks!

### CODE REVIEW

File: src/config/env_directive.rs [hunk 1]
```diff
@@ -79,6 +137,8 @@ impl EnvResults {
     ) -> eyre::Result<Self> {
         let settings = Settings::get();
         let mut ctx = BASE_CONTEXT.clone();
+        debug!("resolve: input: {:#?}", &input);
```
Comment (jdx) [inline src/config/env_directive.rs]: these should all be trace level

Comment (josb) [reply src/config/env_directive.rs]: Will fix, I just added these while trying to understand the code. If you feel some serve little purpose I am happy to remove them.

Also, if you have suggestions for better identifier names, I will happily change them. 🙏🏻

Comment (josb) [reply src/config/env_directive.rs]: I changed all debug calls to trace, and added a custom deserializer to get both the tests to pass and the lazy path entries to be parsed correctly.

**Comment by josb** [comment]
Any feedback, @jdx?

**Pull request merged**